### PR TITLE
MODINVOICE-116 Update purchase order number pattern

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
https://issues.folio.org/browse/MODINVOICE-116

## Approach
* updated poNumbers pattern
* removed repeatable declaration of the uuid pattern

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Examples exist for all schemas
  - [ ] Descriptions exist for all schema properties
  - [ ] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-invoice-storage/schemas/invoice.json [hunk 1]
```diff
@@ -139,14 +135,12 @@
       "type": "array",
       "items": {
         "description": "A human readable ID assigned to this purchase order",
-        "type": "string",
-        "pattern": "^[a-zA-Z0-9]{5,16}$"
+        "$ref": "../../common/schemas/uuid.json"
```
Comment (KVupp) [inline mod-invoice-storage/schemas/invoice.json]: This has to be the poNumber pattern

Comment (grabsefx) [reply mod-invoice-storage/schemas/invoice.json]: Yeah. Actually, only one symbol should be updated by this PR and I have made a mistake :) Fixed

File: mod-invoice-storage/schemas/invoice.json [hunk 2]
```diff
@@ -139,14 +135,12 @@
       "type": "array",
       "items": {
         "description": "A human readable ID assigned to this purchase order",
-        "type": "string",
-        "pattern": "^[a-zA-Z0-9]{5,16}$"
+        "pattern": "^[a-zA-Z0-9]{1,16}$"
```
Comment (KVupp) [inline mod-invoice-storage/schemas/invoice.json]: Also String type, otherwise it will be generated as object

**Comment by grabsefx** [comment]
Updated pattern for receiving_history.json as well

**Pull request merged**